File: codewars/cpp/ipv4_to_int32.cpp
// https://www.codewars.com/kata/52ea928a1ef5cfec800003ee/train/cpp


#include <iostream>
#include <cstdint>
#include <string>
#include <vector>
#include <cmath>

uint32_t ip_to_int32(const std::string& ip) {
    std::vector<uint32_t> digits;
    std::string tmp = "";

    for(auto& i: ip) {
        if(i != '.') {
            tmp += i;
        } else {
            digits.push_back(stoi(tmp));
            tmp = "";
        }
    }
    digits.push_back(stoi(tmp));

    uint32_t int32 = 0;
    int pos = digits.size() - 1;

    for(int i = 0; i < digits.size(); ++i) {
        int32 += digits[i] * pow(256, pos);
        --pos;
    }

    return int32;
}

int main() {
    std::string ip;

    std::cin >> ip;
    std::cout << ip_to_int32(ip) << '\n';

    return 0;
}

File: codewars/python/ipv4_to_int32.py
#!/usr/bin/env python3

# https://www.codewars.com/kata/52ea928a1ef5cfec800003ee/train/python

def ip_to_int32(ip):
    return sum([(pow(256, i[0]) * i[1]) for i in enumerate(reversed([int(i) for i in ip.split('.')]))])

print(ip_to_int32("128.114.17.104"))
print(ip_to_int32("0.0.0.0"))
print(ip_to_int32("128.32.10.1"))
